### PR TITLE
0.6.0 Support for conversion to Stacks and multi versioning

### PR DESCRIPTION
**Pull request by davco01a**
0.6.0 Support for conversion to Stacks and multi versioning

### CODE REVIEW

File: src/main/java/application/rest/StacksAccess.java [hunk 1]
```diff
@@ -0,0 +1,762 @@
...
+				System.out.println("*** " + "Stack name: " + name + " 404 not found");
+				msg.put("status", "Stack name: " + name + " 404 not found");
+				return Response.status(400).entity(msg).build();
+			}
+			List<StackSpecVersions> kabSpecVersions=null;
+			if (version!=null) {
+				kabSpecVersions=StackUtils.getKabInstanceVersions(fromKabanero, name);
+
+				for (StackSpecVersions versionFromKab:kabSpecVersions) {
+					if (version.contentEquals(versionFromKab.getVersion())) {
```
Comment (kaczyns) [inline src/main/java/application/rest/StacksAccess.java]: Do you need to print an error message if they supplied a version, but that version was not currently in the Stack CR spec?

File: src/main/java/application/rest/StacksAccess.java [hunk 2]
```diff
@@ -0,0 +1,762 @@
...
+				JSONObject resp = new JSONObject();
+				resp.put("message", "your login token has expired, please login again");
+				return Response.status(401).entity(resp).build();
+			}
+			
+			kab = StackUtils.getKabaneroForNamespace(namespace);
+			ArrayList<StackSpecPipelines> pipelines = new ArrayList<StackSpecPipelines>();
+			try {
+				List<KabaneroSpecStacksPipelines> defaultPipelines = kab.getSpec().getStacks().getPipelines();
+				for (KabaneroSpecStacksPipelines defaultPipelineElement: defaultPipelines) {
```
[the remark this answers is not in the document]
Comment (davco01a) [reply src/main/java/application/rest/StacksAccess.java]: thanks, that would be excellent Tim

File: src/main/java/application/rest/StacksAccess.java [hunk 3]
```diff
@@ -0,0 +1,762 @@
...
+		// iterate over new collections and create them
+		try {
+			for (Stack s  : multiVersionNewStacks) {
+				int i=0;
+				Map m=(Map) newStacks.get(i);
+				String updateType="";
+				try {
+					KabaneroApi kApi = new KabaneroApi(apiClient);
+					V1OwnerReference owner = kApi.createOwnerReference(kab);
+					owner.setKind(kab.getKind());
```
[the remark this answers is not in the document]
Comment (davco01a) [reply src/main/java/application/rest/StacksAccess.java]: ok

File: src/main/java/application/rest/StacksAccess.java (hunk 1 above)
Comment (davco01a) [reply src/main/java/application/rest/StacksAccess.java]: yes, good idea

File: src/main/java/application/rest/StackUtils.java [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+import io.kubernetes.client.apis.CoreV1Api;
+import io.kubernetes.client.models.V1ContainerStatus;
+import io.kubernetes.client.models.V1Pod;
+import io.kubernetes.client.models.V1PodList;
+import io.kubernetes.client.models.V1PodStatus;
+import io.kubernetes.client.proto.Meta.Status;
+
+public class StackUtils {
+	
+	public static boolean readGitSuccess=true;
```
Comment (brianxjx) [inline src/main/java/application/rest/StackUtils.java]: should this initially be false ?

Comment (tseelbach) [reply src/main/java/application/rest/StackUtils.java]: Dave will consider and probably change in upcoming version

**Pull request merged**